libs/sdk-core/src/breez_services.rs

```diff
@@ -1154,6 +1156,57 @@ impl BreezServices {
 
         Ok(())
     }
+
+    async fn closed_channel_to_transaction(
+        &self,
+        channel: crate::models::Channel,
+    ) -> Result<Payment> {
+        let now_epoch_sec = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
+
+        // If closed_at is missing (older DB where field was not yet part of the schema),
+        // we get it from the channel closing tx
+        let channel_closed_at = channel.closed_at.unwrap_or(
+            {
+                // Lookup tx based on channel funding tx ID
+                let outspends = self
+                    .chain_service
+                    .transaction_outspends(channel.funding_txid.clone())
+                    .await?;
+
+                // If outputs are spent, take the block time as the channel closing timestamp
+                match outspends.iter().find(|&out| out.spent) {
```

Comment:
The funding tx may have other output. We need to find the one that funded the channel and here we only take the first one.

Reply:
Maybe we can select the output that matches the channel capacity?

Reply:
Finding the output based on `funding_outnum`, as discussed.